### PR TITLE
chore: preserve non-org dashboard for platform users

### PR DESCRIPTION
**Pull request by ThyMinimalDev**
## What does this PR do?

Preserve regular dashboard for platform users since they don't have custom domains

https://github.com/user-attachments/assets/be33f51e-b6c6-4db0-985a-e02321b36829

In order to preserve the dashboard with minimal change, we try to exclude the platform org profiles when it's not necessary

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] N/A - I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- create a platform org
- try to access public profile page
- it should still point to /username
- /platform/settings should be accessible

### CODE REVIEW

**Review by supalarry — DISMISSED** [review]
🔥

**Review by hariombalhara — DISMISSED** [review]
Looks like a clean solution but it will need some more work to ensure all links are working fine. Because we use other endpoints too to identify the organization, we will have to handle there as well.

I am not sure of a clean approach to handle it but it would be messy to make things behave like not an org when it is an org technically.

**EventType Booking page isn't working:** https://www.loom.com/share/62183473398f4027a2d1d49b60490634

I am fine merging it and taking care of other things in a followup though.

**Comment by ThyMinimalDev** [comment]
> Looks like a clean solution but it will need some more work to ensure all links are working fine. Because we use other endpoints too to identify the organization, we will have to handle there as well.
> 
> I am not sure of a clean approach to handle it but it would be messy to make things behave like not an org when it is an org technically.
> 
> **EventType Booking page isn't working:** https://www.loom.com/share/62183473398f4027a2d1d49b60490634
> 
> I am fine merging it and taking care of other things in a followup though.

Hey Hariom, I have a new commit that I will push soon hopinh to fix all the links at once

File: packages/ui/components/icon/icon-list.mjs [hunk 1]
```diff
@@ -136,4 +136,5 @@ export const lucideIconList = new Set([
   "webhook",
   "x",
   "zap",
+  "blocks",
 ]);
```
Comment (zomars) [inline packages/ui/components/icon/icon-list.mjs:140]: @ThyMinimalDev lets sort this alphabetically as well 🙏

**Review by zomars — APPROVED** [review]
LGTM. left a non blocking change

File: packages/features/eventtypes/lib/getPublicEvent.ts [hunk 2]
```diff
@@ -248,14 +248,35 @@ export const getPublicEvent = async (
...
-  const event = await prisma.eventType.findFirst({
+  let event = await prisma.eventType.findFirst({
     where: {
       slug: eventSlug,
       ...usersOrTeamQuery,
     },
     select: publicEventSelect,
   });
 
+  // If no event was found, check for platform org user event
```
Comment (hariombalhara) [inline packages/features/eventtypes/lib/getPublicEvent.ts:259]: // If no event was found, check for platform org's unmanaged user 's events

**Review by hariombalhara — COMMENTED** [review]
Looks good

**Pull request merged**